data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/SqsWorker.java

```diff
@@ -67,10 +67,11 @@ public class SqsWorker implements Runnable {
     private final Timer sqsMessageDelayTimer;
     private final Backoff standardBackoff;
     private int failedAttemptCount;
-    private boolean endToEndAcknowledgementsEnabled;
+    private final boolean endToEndAcknowledgementsEnabled;
     private final AcknowledgementSetManager acknowledgementSetManager;
 
     private final ObjectMapper objectMapper = new ObjectMapper();
+    private boolean isStopped;
```

Comment:
We should probably make a package protected setter and call it when the `stop()` method is called on the processor. Right now this is based on catching an interrupt exception, but this might be thrown during any point in the while loop.

Also, please mark `isStopped` as volatile.